Tests/SentryTests/SentryCrash/TestSentryCrashAdapter.m

```diff
@@ -0,0 +1,30 @@
+#import "TestSentryCrashAdapter.h"
+#import <Foundation/Foundation.h>
+
+@implementation TestSentryCrashAdapter
+
++ (instancetype)sharedInstance
```

Comment:
M: as this is just a test class, it does not need to be a singleton right, actually, for testing, it'd be even better if not I guess, so you don't have to reset the fields, this would make testability deterministic probably.
unless you have to because of the interface signature.

Reply:
See my other comment
```
/**
 * Written in Objective-C because Swift doesn't allow you to call the constructor of
 * TestSentryCrashAdapter.
 */
```

We are not able to do `TestSentryCrashAdapter()` in Swift, because of `SENTRY_NO_INIT` in `SentryCrashAdapter`. That's why I need to override the `sharedInstance`.